engine/src/main/java/org/terasology/rendering/cameras/Camera.java

```diff
@@ -205,17 +207,14 @@ public Vector3f getViewingDirection() {
      * Get the orientation of the camera.
      * @return the orientation direction, a quaternion.
      */
-    public Quat4f getOrientation() {
-        return new Quat4f(JomlUtil.from(viewingDirection), viewingAngle);
+    public Quaternionf getOrientation() {
+        return new Quaternionf().fromAxisAngleRad(viewingDirection, viewingAngle);
     }
 
-    /**
-     Try to set the viewing direction.
-     * @param direction
-     */
-    public void setOrientation(Quat4f direction) {
-        viewingDirection = new Vector3f(direction.x,direction.y,direction.z);
-        viewingAngle = direction.getAngle();
+    public void setOrientation(Quaternionf quaternion) {
+        AxisAngle4f axisAngle = new AxisAngle4f(quaternion);
+        viewingDirection.set(axisAngle.x, axisAngle.y, axisAngle.z);
+        viewingAngle = axisAngle.angle;
```

Comment:
Note that the old implementation wrongfully interpreted the provided quaternion as an AxisAngle, which is mathematically wrong. I fixed this, but client code needs to be reviewed urgently.